src/lsp.hh

```diff
@@ -31,7 +32,7 @@ struct RequestId {
   enum Type { kNone, kInt, kString };
```

Comment:
std::variant might be a good alternative

Reply:
std::variant was used but removed in commit "Remove variant and clean up". If it is to be used in more places, it can be reconsidered.